syntax = "proto3";

package chefmoji;

message MapUpdate {
    // A global game synchronizing tick count may be used in the future.
    //
    // In format [ROW1_STR,ROW2_STR,ROW3_STR].
    repeated string map = 1;
    repeated PlayerUpdate players = 2; 
    repeated OrderUpdate order = 3;
}

message PlayerUpdate {
    // In format (MapX, MapY). X goes from left to right, and Y top to bottom of the map.
    repeated uint32 position = 1;
    string inventory = 2;
}

enum OrderType {
    HOT_DOG = 1;
    PIZZA = 2;
    TACO = 3;
    GYRO = 4;
    SANDWICH = 5;
    HAMBURGER = 6;
    WAFFLES = 7;
    PANCAKES = 8;
    EGGS = 9;
    BURRITO = 10;
    SUSHI = 11;
    RAMEN = 12;
    BENTO_BOX = 13;
    STEW = 14;
    CURRY_RICE = 15;
}

message OrderUpdate {
    uint32 uid = 1;
    // in Unix Epoch Seconds
    uint32 registration_time = 2;
    /* The OrderType is the type of food produced by cooking the recipe
       There is no need to provide a list of ingredients for each recipe 
       because both client and server will agree upon the recipe for each
       order beforehand */
    OrderType order_type = 3;
    // Server To Client: This order has been fulfilled. Tear it down.
    bool fulfilled = 4;
}